### PR TITLE
Official Docker image and documentation

### PR DESCRIPTION
**Pull request by JDiPierro**
For #849 :)

Adds a basic `Dockerfile` that starts from `python:3.6` and a includes custom startup script that allows you to start in any mode.

Includes a first attempt at documentation.

I added a Travis stage to build the image but for releasing I think the better approach is for someone to create a DockerHub account for locustio and set up an [automated build](https://docs.docker.com/docker-hub/builds/) for the repository. I believe they do security scanning on the resulting image as well.

Let me know if there's any docs changes, extra features, etc you'd like to see or if I can help with the DockerHub setup. I'm hanging around the Slack server as @JDiPierro as well :D

### CODE REVIEW

**Comment by cgoldberg** [comment]
thanks for this!  
I'd like a few maintainers to give this a review and try it out, so we are able to maintain it and answer questions that come up.
@aldenpeterson-wf @mbeacom can you guys give this a look and try it out when you have a chance?

**Comment by heyman** [comment]
Hey! I think having an official Docker image for Locust would be great.

Do we really need the `docker_start.sh` shell-script? It introduces a new API and interface that you have to "learn" to use Locust with Docker. Couldn't we just set the Docker entrypoint to the locust binary?

File: docker_start.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+if [ -z "${TARGET_URL}" ]; then
```
Comment (mbeacom) [inline docker_start.sh]: While I agree we should handle env vars for all of these settings, I think we should handle these at the app leveland simply take whatever the `$@` entrypoint is, defaulting to `locust`.  This allows the user to modify their entry as necessary and simply pass the envvars to be used.

File: docs/running-locust-docker.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Add your tests
+---------------------------------------------
+
+The easiest way to get your tests running is to build an image with your test file built in. Once you've 
+written your locustfile you can bake it into a Docker image with a simple ``Dockerfile``:
+
+```
+FROM locustio/locust
+ADD locustfile.py locustfile.py
```
Comment (mbeacom) [inline docs/running-locust-docker.rst]: Maybe we should discourage the usage of `ADD` for the user's `locustfile.py` and document mounting a directory with the `locustfile.py` instead.

File: examples/docker-compose/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+version: "3.4"
```
Comment (mbeacom) [inline examples/docker-compose/docker-compose.yml]: I'm all for using the latest `docker-compose` syntax version, but maybe we could pin at `3` to increase availability/adoption.

**Comment by sekka1** [comment]
I have tried the docker image as is in this PR branch and it works great!  There was a Locust Helm chart in the stable repository that hasn't been updated in about a year.  I made all of the changes necessary to do the update to v0.9.0 and also added a bunch of other Kubernetes items in there to make it work a little nicer like nodeSelectors/Taints and an Ingress.  

I have this PR for the Helm Chart update but I should probably wait for an official image instead of using the image in my Docker Hub repo.  Here is the updated Chart: https://github.com/helm/charts/compare/master...sekka1:with-v0.9.0?expand=1#diff-c9789cc33518b31f9230b6284b2b6116R4

The last response to the updates was about a month ago.  I'm happy to continue on if @JDiPierro is too busy now.  

Thoughs @mbeacom

**Comment by cgoldberg** [comment]
> I'm happy to continue on if @JDiPierro is too busy now.

@sekka1 this PR was superseded by #882 , which has already been merged... so Docker stuff is in master branch now.   Could you give it a try?

I am going to close this one now.

**Comment by sekka1** [comment]
@cgoldberg Is there any plans for an official Locust Docker hub image?

**Comment by quorak** [comment]
It would be great to release the image to an official account. This account looks like scam https://hub.docker.com/r/locustio/locust , though not sure if it actually is.

https://hub.docker.com/r/_/locust would be great !

**Comment by swarupdonepudi** [comment]
Is this docker image built and pushed to a public registry yet?

**Comment by swarupdonepudi** [comment]
> Is this docker image built and pushed to a public registry yet?

Nevermind. There is an ongoing discussion around this topic - https://github.com/locustio/locust/issues/849

**Pull request closed without merging**